roachtest: activerecord failed

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18228342?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18228342?buildTab=artifacts#/activerecord) on release-24.2 @ [67fbab57387089a3ac07086f024960ecf51c4824](https://github.com/cockroachdb/cockroach/commits/67fbab57387089a3ac07086f024960ecf51c4824):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v24.2.7-dev-67fbab57387089a3ac07086f024960ecf51c4824
Tests run against activerecord 7.2.1
8486 Total Tests Run
8461 tests passed
25 tests failed
60 tests skipped
6 tests ignored
0 tests passed unexpectedly
2 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: TestAutosaveAssociationOnAHasManyAssociation#test_should_not_update_children_when_parent_creation_with_no_reason (unexpected)
--- FAIL: TestAutosaveAssociationOnAHasManyAssociation#test_should_update_children_when_autosave_is_true_and_parent_is_new_but_child_is_not (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18228342/activerecord/1734504568936/1734513399114)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.6-rc]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.3.1-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45690

--- a/pkg/cmd/roachtest/tests/activerecord_blocklist.go
+++ b/pkg/cmd/roachtest/tests/activerecord_blocklist.go
@@ -22,10 +22,14 @@ var activeRecordBlocklist = blocklist{
 }
 
 var activeRecordIgnoreList = blocklist{
+	`ActiveRecord::CockroachDBStructureDumpTest#test_structure_dump`:                                                                                           "flaky",
 	`ActiveRecord::ConnectionAdapters::PostgreSQLAdapterTest#test_translate_no_connection_exception_to_not_established`:                                        "pg_terminate_backend not implemented",
 	`BasicsTest#test_default_values_are_deeply_dupped`:                                                                                                         "flaky",
 	`CockroachDB::FixturesTest#test_create_fixtures`:                                                                                                           "flaky",
 	`TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_not_save_and_return_false_if_a_callback_cancelled_saving_in_either_create_or_update`: "flaky",
 	`TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_not_update_children_when_parent_creation_with_no_reason`:                             "flaky",
 	`TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_update_children_when_autosave_is_true_and_parent_is_new_but_child_is_not`:            "flaky",
+	`TestAutosaveAssociationOnAHasManyAssociation#test_should_not_save_and_return_false_if_a_callback_cancelled_saving_in_either_create_or_update`:             "flaky",
+	`TestAutosaveAssociationOnAHasManyAssociation#test_should_not_update_children_when_parent_creation_with_no_reason`:                                         "flaky",
+	`TestAutosaveAssociationOnAHasManyAssociation#test_should_update_children_when_autosave_is_true_and_parent_is_new_but_child_is_not`:                        "flaky",
 }

